packages/discovery-react-components/src/components/DocumentPreview/components/PdfHighlight/utils/textBoxMapping/getTextBoxMapping.ts

```diff
@@ -155,13 +156,14 @@ class Source<SourceCell extends TextLayoutCell, TargetCell extends TextLayoutCel
       matchLength: spanLen(matchedSourceSpan),
       markSourceAsMapped: (text: string) => {
         const mappedSource = matchedSourceProvider.getMatch(text);
+        if (mappedSource?.span) {
+          matchedSourceProvider.consume(mappedSource.span);
+        }
         debug('>> target cell %o to source %o', text, mappedSource);
         return mappedSource?.span;
       },
-      markSourceMappedBySpan: (span: TextSpan) => {
-        if (spanLen(span) > 0) {
-          matchedSourceProvider.consume(span);
-        }
+      markAsMapped: () => {
+        matchedSourceProvider.consume(matchedSourceSpan);
       }
```

Comment:
These changes are to consume text in source cell properly. 

Without the first one, a single character in a source could be mapped to multiple characters in target (i.e. bbox in PDF text item). Without the second one, many characters in target couldn't be mapped to source text.

Reply:
Add comments to explain the code.